pkg/client/k8s.go

```diff
@@ -190,7 +190,12 @@ func (k K8sClient) GetOpenEBSNamespace(casType string) (string, error) {
 	if err != nil || len(pods.Items) == 0 {
 		return "", errors.New("unable to determine openebs namespace")
 	}
-	return pods.Items[0].Namespace, nil
+	ns := pods.Items[0].Namespace
```

Comment:
Why is this change required? 

[K8s] Usually `default` namespace is used while creating resources if the resource namespace isn't specified by the client.

<br/>
<br/>
<br/>
<br/>

Pods(are namespaced resources) will have a a non-empty, namespace. Here's an example of a pod in default namespace. 

```bash
➜  $ kubectl get po csi-6447f6bfcf-mch72 -oyaml | grep namespace
  namespace: default
```

Reply:
Okay this is due to fact that the function return namespace of the case types provided to it. 
Ideally it should return error if the cas-types are not present in the cluster doesn't have any namespace.

e.g., I don't have any zfs volumes present in the cluster, but when I call this function to get the namespace of the zfs volume it returns an empty string as the namespace. If it would provided error that "resource is not deployed in the cluster  hence doesn't have any namespace associated with it", then it would have been more desirable thing.

Reply:
> Ideally it should return error if the cas-types are not present in the cluster

Correct, it will return `"", err` if the cas-type is absent in a cluster, i.e. if the ZFS CSI provisioner is not present, it'd likely return a `non nil error` or at least the len(pods.Items) will be zero(check the conditional statement in line#190).

Have you seen it otherwise here(in that case there'd be a bug in the code which we'd have to fix).

But if `pods.Items[0]` is actually a pod, we'd get a non `""` namespace for sure.

Reply:
One alternative way to think of it might be to change the return type of the function to `(*string, error)`, i.e. getting a different zero-value. In Golang, it's a bit common to not use the returned value(assume it to be a zero-value) if the error is non-nil.